packages/strapi-utils/lib/logger.js

```diff
@@ -4,18 +4,52 @@
  * Logger.
  */
 
-const pino = require('pino');
+
+const fs = require('fs');
+const path = require('path');
 const _ = require('lodash');
+const pino = require('pino');
 
 const logLevels = ['fatal', 'error', 'warn', 'info', 'debug', 'trace'];
 
 function getLogLevel() {
-  if (_.isString(process.env.STRAPI_LOG_LEVEL) && _.includes(logLevels, process.env.STRAPI_LOG_LEVEL.toLowerCase())) {
+  if (
+    _.isString(process.env.STRAPI_LOG_LEVEL) &&
+    _.includes(logLevels, process.env.STRAPI_LOG_LEVEL.toLowerCase())
+  ) {
     return process.env.STRAPI_LOG_LEVEL;
   }
   return 'debug';
 }
 
+const logTypes = ['stdout', 'file'];
+function getLogType() {
+  if (
+    _.isString(process.env.STRAPI_LOG_TYPE) &&
+    _.includes(logTypes, process.env.STRAPI_LOG_TYPE.toLowerCase())
+  ) {
+    return process.env.STRAPI_LOG_TYPE;
+  }
+  return 'stdout';
+}
+
+function parseLogPath(defaultLogFilename = 'strapi.log') {
```

Comment:
You are declaring an argument that never pass in the invocation and looking for an value outside the scope of the function... it doesn't feel good.

In any case, you could let the argument and use the environment variable in the invocation if is needed; let the function do just **one** thing.

Reply:
Hi @maturanomx,

I just wrote the function getLogType()  like that because I wanted to be homogeneous with other old existing functions. 

Do you mean I have to update the whole code (legacy + new) to solve the improvement you described above ?

Reply:
You don't **have to** improve code that's now yours; you can, and it will be appreciate, but taking care of your own code should be enough.